opm/autodiff/SimulatorFullyImplicitBlackoilOutput.cpp

```diff
@@ -248,5 +250,88 @@ namespace Opm
         if ( eclWriter_ ) {
             eclWriter_->writeTimeStep(timer, state, wellState);
         }
+        // write backup file
+        if( backupfile_ )
+        {
+            int reportStep      = timer.reportStepNum();
+            int currentTimeStep = timer.currentStepNum();
+            if( (reportStep == currentTimeStep || // true for SimulatorTimer
+                 currentTimeStep == 0 || // true for AdaptiveSimulatorTimer at reportStep
+                 timer.done() ) // true for AdaptiveSimulatorTimer at reportStep
+               && lastBackupReportStep_ != reportStep ) // only backup report step once
+            {
+                // store report step
+                lastBackupReportStep_ = reportStep;
+                // write resport step number
+                backupfile_.write( (const char *) &reportStep, sizeof(int) );
+
+                const BlackoilState* boState = dynamic_cast< const BlackoilState* > (&state);
+                if( boState ) {
+                    backupfile_ << (*boState);
+                }
+                else
```

Comment:
I have not seen a reply to this (neither the nitpick nor the worry) yet, will fix or not?